### PR TITLE
[go] Update incompatible version screen with links to `expo.dev/go`

### PR DESCRIPTION
**Pull request by behenate**
# Why

We want to link people to `https://expo.dev/go` when they launch an incompatible project in Expo Go so it's easier for them to install a compatible Expo Go app.

ENG-11732

# How

Edited the existing messages to include installation links. Depending if running on a iOS device or in a simulator the behaviour is slightly different.

Also updated the code to work correctly when only one SDK version is available, on iOS added a more universal link parser that changes any combination of `[link](description)` into a hyperlink instead.

# Test Plan

Tested in Expo Go on a physical Android device, emulator, physical iOS device and iOS emulator

|Android|iOS Simulator|iOS Device|
| --- | --- | --- |
|android | simulator| physical|

### CODE REVIEW

**Review by tsapeta — APPROVED** [review]

**Review by kadikraman — APPROVED** [review]
Nice! 🎉

**Comment by brentvatne** [comment]
nice work! what do you think about this for iOS?

image

(I didn't mock this up for Android, let me know if that's needed - we should follow the same pattern)

**Comment by brentvatne** [comment]
another thought - should we remove the "Try again" button here on this specific error screen? it really catches the eye and it's not very useful.

**Comment by behenate** [comment]
Updated version:
|Android|iOS Simulator|iOS Device|
| --- | --- | --- |
|android | simulator| physical|

File: apps/expo-go/ios/Exponent/Kernel/AppLoader/CachedResource/EXManifestResource.m [hunk 1]
```diff
@@ -202,13 +204,31 @@ - (BOOL)_isManifestVerificationBypassed: (id) manifestObj
 
 - (NSInteger)sdkVersionStringToInt:(nonnull NSString *)sdkVersion {
   NSRange snackSdkVersionRange = [sdkVersion rangeOfString: @"."];
-  return [[sdkVersion substringToIndex: snackSdkVersionRange.location] intValue];
+  return [[sdkVersion substringToIndex:snackSdkVersionRange.location] intValue];
 }
 
-- (NSString *)supportedSdkVersionsConjunctionString:(nonnull NSString *)conjuction {
+-(NSMutableArray *)supportedSdkVersionInts {
```
Comment (tsapeta) [inline apps/expo-go/ios/Exponent/Kernel/AppLoader/CachedResource/EXManifestResource.m:210]: Returning mutable arrays/dicts is not a good practice if there is no strong need to do so.
```suggestion
- (NSArray<NSNumber *> *)supportedSdkVersionInts
{
```

**Review by brentvatne — APPROVED** [review]
- How to Fix This Error -> How to fix this error
- add a bit less spacing under the "How to fix this error" header so it feels grouped with the text below it. maybe no extra spacing, just a normal amount of line spacing that we would use if it was part of the same paragraph

**Comment by brentvatne** [comment]
- currently installed -> installed
- The project you attempted to open -> The project you opened

**Comment by behenate** [comment]
@brentvatne Hopefully the final edit 😅:
|iOS|Android|
| --- | --- | 
|android | simulator|

**Comment by brentvatne** [comment]
looks great! thank you!

**Pull request merged**